[enocean] Discovery of D2-50 devices not working anymore in M3

## Expected Behavior
When using the discovery/search for the enocaen binding it should discover D2-50 devices which are in pairing mode as it did in the release version

## Current Behavior
No device is found anymore, this was working before.

## Steps to Reproduce (for Bugs)
- Start search for enocaen binding
- Put D2-50 device in pairing mode
- Nothing hapens

## Context
I was using enocaen with openhab 3.0.1 where I was able to add D2-50 devices with the enocaen discovery. For this I started the discovery in openhab and pressed the pairing button on my device. The device appeared immediately as "Generic" (since D2-50 is not implemented in the release version).
To get full support for this device type I've updated to 3.1.0.M3 and existing devices are working as expected, but it's not possible anymore to pair/discover any new D2-50 devices.

## Your Environment
- Debian buster
- openjdk 11

## Stacktrace
```
2021-04-21 19:59:40.939 [INFO ] [ernal.transceiver.EnOceanTransceiver] - Received teach in message from 050ADD71
2021-04-21 19:59:40.939 [INFO ] [covery.EnOceanDeviceDiscoveryService] - EnOcean Package discovered, RORG UTE, payload D480FF61000050D2050ADD7100, additional 00FFFFFFFF3500
2021-04-21 19:59:40.940 [ERROR] [ding.enocean.internal.eep.EEPFactory] - Cannot instantiate EEP D2-50-00: null
2021-04-21 19:59:40.941 [ERROR] [ernal.transceiver.EnOceanTransceiver] - Exception in informListeners
java.lang.IllegalArgumentException: java.lang.reflect.InvocationTargetException
        at org.openhab.binding.enocean.internal.eep.EEPFactory.buildEEP(EEPFactory.java:75) ~[bundleFile:?]
        at org.openhab.binding.enocean.internal.eep.EEPFactory.buildEEPFromTeachInERP1(EEPFactory.java:202) ~[bundleFile:?]
        at org.openhab.binding.enocean.internal.discovery.EnOceanDeviceDiscoveryService.packetReceived(EnOceanDeviceDiscoveryService.java:104) ~[bundleFile:?]
        at org.openhab.binding.enocean.internal.transceiver.EnOceanTransceiver.informListeners(EnOceanTransceiver.java:296) [bundleFile:?]
        at org.openhab.binding.enocean.internal.transceiver.EnOceanESP3Transceiver.processMessage(EnOceanESP3Transceiver.java:140) [bundleFile:?]
        at org.openhab.binding.enocean.internal.transceiver.EnOceanTransceiver.receivePackets(EnOceanTransceiver.java:261) [bundleFile:?]
        at org.openhab.binding.enocean.internal.transceiver.EnOceanTransceiver$1.run(EnOceanTransceiver.java:197) [bundleFile:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.reflect.InvocationTargetException
        at jdk.internal.reflect.GeneratedConstructorAccessor151.newInstance(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
        at org.openhab.binding.enocean.internal.eep.EEPFactory.buildEEP(EEPFactory.java:67) ~[bundleFile:?]
        ... 12 more
Caused by: java.lang.IllegalArgumentException
        at org.openhab.binding.enocean.internal.eep.EEP.setData(EEP.java:134) ~[bundleFile:?]
        at org.openhab.binding.enocean.internal.eep.EEP.<init>(EEP.java:70) ~[bundleFile:?]
        at org.openhab.binding.enocean.internal.eep.Base._VLDMessage.<init>(_VLDMessage.java:31) ~[bundleFile:?]
        at org.openhab.binding.enocean.internal.eep.D2_50.D2_50.<init>(D2_50.java:63) ~[bundleFile:?]
        at jdk.internal.reflect.GeneratedConstructorAccessor151.newInstance(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
        at org.openhab.binding.enocean.internal.eep.EEPFactory.buildEEP(EEPFactory.java:67) ~[bundleFile:?]
        ... 12 more
```

--- a/bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/D2_50/D2_50.java
+++ b/bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/D2_50/D2_50.java
@@ -44,6 +44,7 @@ public class D2_50 extends _VLDMessage {
     protected static final byte MT_CONTROL = 0x20;
     protected static final byte MT_BASIC_STATUS = 0x40;
     protected static final byte MT_EXTENDED_STATUS = 0x60; // not yet implemented
+    protected static final byte MT_UNKNOWN_STATUS = (byte) 0x80; // Sent by some systems during teach in
 
     protected static final byte rmtMask = (byte) 0x0f;
     protected static final byte RMT_BASIC_STATUS = 0x00;
@@ -177,7 +178,10 @@ public class D2_50 extends _VLDMessage {
                 return bytes.length == 14;
             case MT_EXTENDED_STATUS: // MT_EXTENDED_STATUS is not yet supported, however return true to avoid Exceptions
                 return true;
+            case MT_UNKNOWN_STATUS:
+                return true;
             default:
+                logger.error("Invalid data, unknown message type: {} ({})", getMessageType(bytes[0]), bytes);
                 return false;
         }
     }

